.github/workflows/test.yml

```
@@ -0,0 +1,245 @@
+      # purpose of testing
+      - name: Point TMPDIR to "tricky" setup
+        run: |
+          if [ -n "${_DL_TMPDIR:-}" ]
+          then echo TMPDIR="${_DL_TMPDIR}" >> "$GITHUB_ENV"
+          fi
+
+      - name: Test installation for user
+        run: sudo pip install --user .
+
+      - name: Report WTF information using system-wide installed version
+        run: datalad wtf
+
+      - name: Run tests
+        run: |
+          : "${PYTEST_SELECTION:=$DEFAULT_PYTEST_SELECTION}"
+          : "${PYTEST_SELECTION_OP:=$DEFAULT_PYTEST_SELECTION_OP}"
+          : "${DATALAD_TESTS_SSH:=$DEFAULT_DATALAD_TESTS_SSH}"
+          : "${DATALAD_LOG_ENV:=$DEFAULT_DATALAD_LOG_ENV}"
+
+          PYTEST_OPTS=( -v )
+          # If we requested to run only not slow (typically <10sec) tests, fail
+          # if a test takes 3x more than that - it needs to get @slow or
+          # @turtle annotation
+          if echo "$PYTEST_SELECTION_OP($PYTEST_SELECTION)" | grep -q "^not.*slow"
+          then
+            PYTEST_OPTS=( "${PYTEST_OPTS[@]}" --doctest-modules --durations=0 --durations-min=5 --fail-slow 60 )
+            export DATALAD_TESTS_SETUP_TESTREPOS=1
+          fi
+
+          mkdir -p __testhome__
+          cd __testhome__
+          # Note: adding --log-cli-level=INFO would result in
+          # DATALAD_LOG_TARGET=/dev/null being not in effect, dumping too many
+          # logs.
+          set -x
+          http_proxy=
+          PATH=$PWD/../tools/coverage-bin:$PATH
+          $PYTEST_WRAPPER python \
+            -m pytest "${PYTEST_OPTS[@]}" \
+            -c ../tox.ini \
+            -n 2 \
+            -m "${PYTEST_SELECTION:+$PYTEST_SELECTION_OP($PYTEST_SELECTION) and }not(turtle)" \
+            --doctest-modules \
+            --cov=datalad \
+            --cov-report=xml \
+            --pyargs ${TESTS_TO_PERFORM:-datalad}
+
+      # Makes it only more difficult to comprehend the failing output.  Enable
+      # only when necessary for a particular debugging.
+      - name: Dump debug info
+        #if: "failure()"  # Enabled
+        if: "failure() && false"  # Disabled
+        run: |
+          if [ ! -z "$DATALAD_TESTS_NONETWORK" ]
+          then sudo route add -net 0.0.0.0 netmask 0.0.0.0 dev lo
+          fi
+
+          DATALAD_LOG_LEVEL=DEBUG \
+            $PYTEST_WRAPPER `which pytest` \
+            -s -v --doctest-modules --cov datalad --log-cli-level=DEBUG
+
+          if [ ! -z "$DATALAD_TESTS_NONETWORK" ]
+          then sudo route del -net 0.0.0.0 netmask 0.0.0.0 dev lo
+          fi
+
+      # cron jobs test more and then PRs will be falling behind since they
+      # would not trigger some codepaths.  So submit coverage only from
+      # non-cron jobs, but report for all
+
+      - name: Report coverage
+        run: python -m coverage report
+        working-directory: __testhome__
+
+      - name: Upload coverage to Codecov
+        if: github.event_name == 'schedule'
```

the opposite needed:

```suggestion
        if: github.event_name != 'schedule'
```